src/plugins/vis_type_timeseries/public/application/components/lib/create_select_handler.ts

```diff
@@ -16,24 +16,34 @@
  * specific language governing permissions and limitations
  * under the License.
  */
+import { EuiComboBoxOptionOption } from '@elastic/eui';
 
-import { createSelectHandler } from './create_select_handler';
+export type HandleChange = (partialModel: Record<string, any>) => void;
 
-describe('createSelectHandler()', () => {
-  let handleChange;
-  let changeHandler;
-
-  beforeEach(() => {
-    handleChange = jest.fn();
-    changeHandler = createSelectHandler(handleChange);
-    const fn = changeHandler('test');
-    fn([{ value: 'foo' }]);
+export const createSelectHandler = (handleChange: HandleChange) => (name: string) => (
+  selected: EuiComboBoxOptionOption[] = []
+) =>
+  handleChange?.({
+    [name]: selected[0]?.value ?? null,
   });
 
-  test('calls handleChange() function with partial', () => {
-    expect(handleChange.mock.calls.length).toEqual(1);
-    expect(handleChange.mock.calls[0][0]).toEqual({
-      test: 'foo',
-    });
+export const createCustomLabelSelectHandler = (handleChange: HandleChange) => (name: string) => (
```

Comment:
What if I ask you for a favour: 
Could you please get rid of `createCustomLabelSelectHandler` at all and move the callback directly into `FieldSelect` ?
This will help us to get rid of bunch of wrappers such as `handleFieldsChange` and extra re-renders of the `FieldSelect` component, which currently re-renders each time of parent render, since the prop `onChange` is a new function each time..